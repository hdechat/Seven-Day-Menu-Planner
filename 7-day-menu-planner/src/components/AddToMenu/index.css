.add-to-menu {
  margin: 0.3rem;
}

.select-category {
  border: solid 1px;
  margin: auto;
  width: 9rem;
}

select {
  background: transparent;
  background-image: none;
  border: none;
  height: 2rem;
  outline: none;
  padding: 5px 8px;
}

p {
  margin-bottom: 0rem;
}

button {
  height: 2rem;
  widows: 9rem;
}