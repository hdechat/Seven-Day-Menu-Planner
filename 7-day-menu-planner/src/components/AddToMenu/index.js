import React, { Component } from 'react';
import './index.css';

export default class AddToMenu extends Component {
  constructor(props) {
    super(props);

    this.state = {
      weekday: '',
      mealTime: ''
    };
  }

  handleClick = (event) => {
    this.setState({ [event.target.name]: event.target.value});
  }

  handleSubmit = (event) => {
    const { 
      addToMenu, addToGroceryList, 
      recipe, addGroceryListToStorage } = this.props;

    event.preventDefault();
    addToMenu(this.state.weekday, this.state.mealTime, this.props.recipe);
    addToGroceryList(recipe.ingredients);
    addGroceryListToStorage(recipe.ingredients);
  }

  listWeekdayOptions = () => {
    const weekdays = [
      'sunday', 'monday', 'tuesday', 
      'wednesday', 'thursday', 'friday', 'saturday'
    ];

    return weekdays.map(day => <option key={day} value={day}>{day}</option>);
  }

  listMealTimeOptions = () => {
    const mealTimes = ['breakfast', 'lunch', 'dinner'];

    return mealTimes.map(meal => <option key={meal} value={meal}>{meal}</option>);
  }

  render() {
    return (
      <div>
        <p>Add this meal to your menu!</p>
        <form className="add-to-menu" onSubmit={this.handleSubmit}>
          <div className="select-category">
            <select name="weekday" id="weekday" onChange={this.handleClick}>
              <option>SELECT WEEKDAY</option>
              {this.listWeekdayOptions()}
            </select>
          </div>
          <div className="select-category">
            <select name="mealTime" id="meal-time" onChange={this.handleClick}>
              <option>SELECT MEAL-TIME</option>
              {this.listMealTimeOptions()}
            </select>
          </div>
          <button type='submit'>Add To Menu</button>
        </form>
      </div>
    );
  }
}
