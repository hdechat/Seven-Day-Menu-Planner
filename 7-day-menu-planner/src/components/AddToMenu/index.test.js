import React from 'react';
import { shallow, mount } from 'enzyme';
import AddToMenu from './index.js';

describe('Add To Menu', () => {
  let wrapper;
  let mockRecipe;
  let mockAddToMenu;
  let mockAddToGroceryList;

  beforeEach(() => {
    mockRecipe = {
      "image": "https://www.edamam.com/web-img/e50/e50584043c015a91a26a7",
      "ingredients": [
        "1½ pounds fresh red jalapeño peppers, stemmed, seeded, and roughly chopped", 
        "8 garlic cloves, peeled and smashed",
         "⅓ cup apple cider vinegar", 
         "3 tablespoons tomato paste", 
         "3 tablespoons honey", 
         "2 tablespoons Paleo-friendly fish sauce (Red Boat!)", 
         "1½ teaspoons kosher salt"
       ],
      "title": "Paleo Sriracha recipes",
      "url": "http://nomnompaleo.com/post/36060636540/paleo-sriracha"
    };

    mockAddToMenu = jest.fn();
    mockAddToGroceryList = jest.fn();

    wrapper = shallow(<AddToMenu 
      addToGroceryList={mockAddToGroceryList} 
      addToMenu={mockAddToMenu} 
      recipe={mockRecipe}/>);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Select Weekday', () => {
    let mockEvent;

    beforeEach(() => {
      mockEvent = { target: {
        name: 'weekday',
        value: 'Monday'
      }};
    });

    it('should call handleClick when user selects weekday', () => {
      wrapper = mount(<AddToMenu 
        addToGroceryList={jest.fn()} 
        addToMenu={jest.fn()} 
        recipe={mockRecipe} />);

      const spy = spyOn(wrapper.instance(), 'handleClick');

      wrapper.instance().forceUpdate();
      wrapper.find('#weekday').simulate('change', mockEvent);

      expect(spy).toHaveBeenCalled();
    });

    it('should update weekday in state when weekday is selected', () => {
      wrapper.instance().handleClick(mockEvent);

      expect(wrapper.state('weekday')).toEqual('Monday');
    });
  });

  describe('Select MealTime', () => {
    let mockEvent;

    beforeEach(() => {
      mockEvent = { target: {
        name: 'mealTime',
        value: 'Breakfast'
      }};
    });

    it('should call handleClick when user selects meal time', () => {
      wrapper = mount(<AddToMenu 
        addToGroceryList={jest.fn()} 
        addToMenu={jest.fn()} 
        recipe={mockRecipe} />);

      const spy = spyOn(wrapper.instance(), 'handleClick');

      wrapper.instance().forceUpdate();
      wrapper.find('#meal-time').simulate('change', mockEvent);

      expect(spy).toHaveBeenCalled();
    });

    it('should update weekday in state when weekday is selected', () => {
      wrapper.instance().handleClick(mockEvent);

      expect(wrapper.state('mealTime')).toEqual('Breakfast');
    });
  });

  it('should call handleSubmit on submit', () => {
    wrapper = mount(<AddToMenu 
      addToGroceryList={jest.fn()} 
      addToMenu={jest.fn()} 
      recipe={mockRecipe} />);

    const spy = spyOn(wrapper.instance(), 'handleSubmit');

    wrapper.instance().forceUpdate();
    wrapper.find('form').simulate('submit');

    expect(spy).toHaveBeenCalled();
  });

  it('should call addToMenu and addToGroceryList when handleSubmit is fired', () => {
    wrapper.instance().handleSubmit({preventDefault: jest.fn()});

    expect(mockAddToMenu).toHaveBeenCalled();
    expect(mockAddToGroceryList).toHaveBeenCalled();
  });
});
