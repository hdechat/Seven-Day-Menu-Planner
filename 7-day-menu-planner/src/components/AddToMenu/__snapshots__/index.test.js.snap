// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add To Menu matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddToMenu
    addGroceryListToStorage={[Function]}
    addToGroceryList={[Function]}
    addToMenu={[Function]}
    recipe={
        Object {
            "image": "https://www.edamam.com/web-img/e50/e50584043c015a91a26a7",
            "ingredients": Array [
              "1½ pounds fresh red jalapeño peppers, stemmed, seeded, and roughly chopped",
              "8 garlic cloves, peeled and smashed",
              "⅓ cup apple cider vinegar",
              "3 tablespoons tomato paste",
              "3 tablespoons honey",
              "2 tablespoons Paleo-friendly fish sauce (Red Boat!)",
              "1½ teaspoons kosher salt",
            ],
            "title": "Paleo Sriracha recipes",
            "url": "http://nomnompaleo.com/post/36060636540/paleo-sriracha",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Add this meal to your menu!
</p>,
        <form
          className="add-to-menu"
          onSubmit={[Function]}
>
          <div
                    className="select-category"
          >
                    <select
                              id="weekday"
                              name="weekday"
                              onChange={[Function]}
                    >
                              <option>
                                        SELECT WEEKDAY
                              </option>
                              <option
                                        value="sunday"
                              >
                                        sunday
                              </option>
                              <option
                                        value="monday"
                              >
                                        monday
                              </option>
                              <option
                                        value="tuesday"
                              >
                                        tuesday
                              </option>
                              <option
                                        value="wednesday"
                              >
                                        wednesday
                              </option>
                              <option
                                        value="thursday"
                              >
                                        thursday
                              </option>
                              <option
                                        value="friday"
                              >
                                        friday
                              </option>
                              <option
                                        value="saturday"
                              >
                                        saturday
                              </option>
                    </select>
          </div>
          <div
                    className="select-category"
          >
                    <select
                              id="meal-time"
                              name="mealTime"
                              onChange={[Function]}
                    >
                              <option>
                                        SELECT MEAL-TIME
                              </option>
                              <option
                                        value="breakfast"
                              >
                                        breakfast
                              </option>
                              <option
                                        value="lunch"
                              >
                                        lunch
                              </option>
                              <option
                                        value="dinner"
                              >
                                        dinner
                              </option>
                    </select>
          </div>
          <button
                    type="submit"
          >
                    Add To Menu
          </button>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add this meal to your menu!",
        },
        "ref": null,
        "rendered": "Add this meal to your menu!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="select-category"
>
              <select
                            id="weekday"
                            name="weekday"
                            onChange={[Function]}
              >
                            <option>
                                          SELECT WEEKDAY
                            </option>
                            <option
                                          value="sunday"
                            >
                                          sunday
                            </option>
                            <option
                                          value="monday"
                            >
                                          monday
                            </option>
                            <option
                                          value="tuesday"
                            >
                                          tuesday
                            </option>
                            <option
                                          value="wednesday"
                            >
                                          wednesday
                            </option>
                            <option
                                          value="thursday"
                            >
                                          thursday
                            </option>
                            <option
                                          value="friday"
                            >
                                          friday
                            </option>
                            <option
                                          value="saturday"
                            >
                                          saturday
                            </option>
              </select>
</div>,
            <div
              className="select-category"
>
              <select
                            id="meal-time"
                            name="mealTime"
                            onChange={[Function]}
              >
                            <option>
                                          SELECT MEAL-TIME
                            </option>
                            <option
                                          value="breakfast"
                            >
                                          breakfast
                            </option>
                            <option
                                          value="lunch"
                            >
                                          lunch
                            </option>
                            <option
                                          value="dinner"
                            >
                                          dinner
                            </option>
              </select>
</div>,
            <button
              type="submit"
>
              Add To Menu
</button>,
          ],
          "className": "add-to-menu",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <select
                id="weekday"
                name="weekday"
                onChange={[Function]}
>
                <option>
                                SELECT WEEKDAY
                </option>
                <option
                                value="sunday"
                >
                                sunday
                </option>
                <option
                                value="monday"
                >
                                monday
                </option>
                <option
                                value="tuesday"
                >
                                tuesday
                </option>
                <option
                                value="wednesday"
                >
                                wednesday
                </option>
                <option
                                value="thursday"
                >
                                thursday
                </option>
                <option
                                value="friday"
                >
                                friday
                </option>
                <option
                                value="saturday"
                >
                                saturday
                </option>
</select>,
              "className": "select-category",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option>
                    SELECT WEEKDAY
</option>,
                  Array [
                    <option
                      value="sunday"
>
                      sunday
</option>,
                    <option
                      value="monday"
>
                      monday
</option>,
                    <option
                      value="tuesday"
>
                      tuesday
</option>,
                    <option
                      value="wednesday"
>
                      wednesday
</option>,
                    <option
                      value="thursday"
>
                      thursday
</option>,
                    <option
                      value="friday"
>
                      friday
</option>,
                    <option
                      value="saturday"
>
                      saturday
</option>,
                  ],
                ],
                "id": "weekday",
                "name": "weekday",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SELECT WEEKDAY",
                  },
                  "ref": null,
                  "rendered": "SELECT WEEKDAY",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "sunday",
                  "nodeType": "host",
                  "props": Object {
                    "children": "sunday",
                    "value": "sunday",
                  },
                  "ref": null,
                  "rendered": "sunday",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "monday",
                  "nodeType": "host",
                  "props": Object {
                    "children": "monday",
                    "value": "monday",
                  },
                  "ref": null,
                  "rendered": "monday",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "tuesday",
                  "nodeType": "host",
                  "props": Object {
                    "children": "tuesday",
                    "value": "tuesday",
                  },
                  "ref": null,
                  "rendered": "tuesday",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "wednesday",
                  "nodeType": "host",
                  "props": Object {
                    "children": "wednesday",
                    "value": "wednesday",
                  },
                  "ref": null,
                  "rendered": "wednesday",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "thursday",
                  "nodeType": "host",
                  "props": Object {
                    "children": "thursday",
                    "value": "thursday",
                  },
                  "ref": null,
                  "rendered": "thursday",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "friday",
                  "nodeType": "host",
                  "props": Object {
                    "children": "friday",
                    "value": "friday",
                  },
                  "ref": null,
                  "rendered": "friday",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "saturday",
                  "nodeType": "host",
                  "props": Object {
                    "children": "saturday",
                    "value": "saturday",
                  },
                  "ref": null,
                  "rendered": "saturday",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <select
                id="meal-time"
                name="mealTime"
                onChange={[Function]}
>
                <option>
                                SELECT MEAL-TIME
                </option>
                <option
                                value="breakfast"
                >
                                breakfast
                </option>
                <option
                                value="lunch"
                >
                                lunch
                </option>
                <option
                                value="dinner"
                >
                                dinner
                </option>
</select>,
              "className": "select-category",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option>
                    SELECT MEAL-TIME
</option>,
                  Array [
                    <option
                      value="breakfast"
>
                      breakfast
</option>,
                    <option
                      value="lunch"
>
                      lunch
</option>,
                    <option
                      value="dinner"
>
                      dinner
</option>,
                  ],
                ],
                "id": "meal-time",
                "name": "mealTime",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SELECT MEAL-TIME",
                  },
                  "ref": null,
                  "rendered": "SELECT MEAL-TIME",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "breakfast",
                  "nodeType": "host",
                  "props": Object {
                    "children": "breakfast",
                    "value": "breakfast",
                  },
                  "ref": null,
                  "rendered": "breakfast",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "lunch",
                  "nodeType": "host",
                  "props": Object {
                    "children": "lunch",
                    "value": "lunch",
                  },
                  "ref": null,
                  "rendered": "lunch",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "dinner",
                  "nodeType": "host",
                  "props": Object {
                    "children": "dinner",
                    "value": "dinner",
                  },
                  "ref": null,
                  "rendered": "dinner",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add To Menu",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Add To Menu",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Add this meal to your menu!
</p>,
          <form
            className="add-to-menu"
            onSubmit={[Function]}
>
            <div
                        className="select-category"
            >
                        <select
                                    id="weekday"
                                    name="weekday"
                                    onChange={[Function]}
                        >
                                    <option>
                                                SELECT WEEKDAY
                                    </option>
                                    <option
                                                value="sunday"
                                    >
                                                sunday
                                    </option>
                                    <option
                                                value="monday"
                                    >
                                                monday
                                    </option>
                                    <option
                                                value="tuesday"
                                    >
                                                tuesday
                                    </option>
                                    <option
                                                value="wednesday"
                                    >
                                                wednesday
                                    </option>
                                    <option
                                                value="thursday"
                                    >
                                                thursday
                                    </option>
                                    <option
                                                value="friday"
                                    >
                                                friday
                                    </option>
                                    <option
                                                value="saturday"
                                    >
                                                saturday
                                    </option>
                        </select>
            </div>
            <div
                        className="select-category"
            >
                        <select
                                    id="meal-time"
                                    name="mealTime"
                                    onChange={[Function]}
                        >
                                    <option>
                                                SELECT MEAL-TIME
                                    </option>
                                    <option
                                                value="breakfast"
                                    >
                                                breakfast
                                    </option>
                                    <option
                                                value="lunch"
                                    >
                                                lunch
                                    </option>
                                    <option
                                                value="dinner"
                                    >
                                                dinner
                                    </option>
                        </select>
            </div>
            <button
                        type="submit"
            >
                        Add To Menu
            </button>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add this meal to your menu!",
          },
          "ref": null,
          "rendered": "Add this meal to your menu!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="select-category"
>
                <select
                                id="weekday"
                                name="weekday"
                                onChange={[Function]}
                >
                                <option>
                                                SELECT WEEKDAY
                                </option>
                                <option
                                                value="sunday"
                                >
                                                sunday
                                </option>
                                <option
                                                value="monday"
                                >
                                                monday
                                </option>
                                <option
                                                value="tuesday"
                                >
                                                tuesday
                                </option>
                                <option
                                                value="wednesday"
                                >
                                                wednesday
                                </option>
                                <option
                                                value="thursday"
                                >
                                                thursday
                                </option>
                                <option
                                                value="friday"
                                >
                                                friday
                                </option>
                                <option
                                                value="saturday"
                                >
                                                saturday
                                </option>
                </select>
</div>,
              <div
                className="select-category"
>
                <select
                                id="meal-time"
                                name="mealTime"
                                onChange={[Function]}
                >
                                <option>
                                                SELECT MEAL-TIME
                                </option>
                                <option
                                                value="breakfast"
                                >
                                                breakfast
                                </option>
                                <option
                                                value="lunch"
                                >
                                                lunch
                                </option>
                                <option
                                                value="dinner"
                                >
                                                dinner
                                </option>
                </select>
</div>,
              <button
                type="submit"
>
                Add To Menu
</button>,
            ],
            "className": "add-to-menu",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <select
                  id="weekday"
                  name="weekday"
                  onChange={[Function]}
>
                  <option>
                                    SELECT WEEKDAY
                  </option>
                  <option
                                    value="sunday"
                  >
                                    sunday
                  </option>
                  <option
                                    value="monday"
                  >
                                    monday
                  </option>
                  <option
                                    value="tuesday"
                  >
                                    tuesday
                  </option>
                  <option
                                    value="wednesday"
                  >
                                    wednesday
                  </option>
                  <option
                                    value="thursday"
                  >
                                    thursday
                  </option>
                  <option
                                    value="friday"
                  >
                                    friday
                  </option>
                  <option
                                    value="saturday"
                  >
                                    saturday
                  </option>
</select>,
                "className": "select-category",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option>
                      SELECT WEEKDAY
</option>,
                    Array [
                      <option
                        value="sunday"
>
                        sunday
</option>,
                      <option
                        value="monday"
>
                        monday
</option>,
                      <option
                        value="tuesday"
>
                        tuesday
</option>,
                      <option
                        value="wednesday"
>
                        wednesday
</option>,
                      <option
                        value="thursday"
>
                        thursday
</option>,
                      <option
                        value="friday"
>
                        friday
</option>,
                      <option
                        value="saturday"
>
                        saturday
</option>,
                    ],
                  ],
                  "id": "weekday",
                  "name": "weekday",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "SELECT WEEKDAY",
                    },
                    "ref": null,
                    "rendered": "SELECT WEEKDAY",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": "sunday",
                    "nodeType": "host",
                    "props": Object {
                      "children": "sunday",
                      "value": "sunday",
                    },
                    "ref": null,
                    "rendered": "sunday",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": "monday",
                    "nodeType": "host",
                    "props": Object {
                      "children": "monday",
                      "value": "monday",
                    },
                    "ref": null,
                    "rendered": "monday",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": "tuesday",
                    "nodeType": "host",
                    "props": Object {
                      "children": "tuesday",
                      "value": "tuesday",
                    },
                    "ref": null,
                    "rendered": "tuesday",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": "wednesday",
                    "nodeType": "host",
                    "props": Object {
                      "children": "wednesday",
                      "value": "wednesday",
                    },
                    "ref": null,
                    "rendered": "wednesday",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": "thursday",
                    "nodeType": "host",
                    "props": Object {
                      "children": "thursday",
                      "value": "thursday",
                    },
                    "ref": null,
                    "rendered": "thursday",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": "friday",
                    "nodeType": "host",
                    "props": Object {
                      "children": "friday",
                      "value": "friday",
                    },
                    "ref": null,
                    "rendered": "friday",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": "saturday",
                    "nodeType": "host",
                    "props": Object {
                      "children": "saturday",
                      "value": "saturday",
                    },
                    "ref": null,
                    "rendered": "saturday",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <select
                  id="meal-time"
                  name="mealTime"
                  onChange={[Function]}
>
                  <option>
                                    SELECT MEAL-TIME
                  </option>
                  <option
                                    value="breakfast"
                  >
                                    breakfast
                  </option>
                  <option
                                    value="lunch"
                  >
                                    lunch
                  </option>
                  <option
                                    value="dinner"
                  >
                                    dinner
                  </option>
</select>,
                "className": "select-category",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option>
                      SELECT MEAL-TIME
</option>,
                    Array [
                      <option
                        value="breakfast"
>
                        breakfast
</option>,
                      <option
                        value="lunch"
>
                        lunch
</option>,
                      <option
                        value="dinner"
>
                        dinner
</option>,
                    ],
                  ],
                  "id": "meal-time",
                  "name": "mealTime",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "SELECT MEAL-TIME",
                    },
                    "ref": null,
                    "rendered": "SELECT MEAL-TIME",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": "breakfast",
                    "nodeType": "host",
                    "props": Object {
                      "children": "breakfast",
                      "value": "breakfast",
                    },
                    "ref": null,
                    "rendered": "breakfast",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": "lunch",
                    "nodeType": "host",
                    "props": Object {
                      "children": "lunch",
                      "value": "lunch",
                    },
                    "ref": null,
                    "rendered": "lunch",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": "dinner",
                    "nodeType": "host",
                    "props": Object {
                      "children": "dinner",
                      "value": "dinner",
                    },
                    "ref": null,
                    "rendered": "dinner",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add To Menu",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Add To Menu",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
